### PR TITLE
New formula checking and reaction balancing

### PR DESCRIPTION
**Pull request by samseaver**
This update is a partial update of sorts, it concerns some debugging of InChI loading/parsing, and it will update formulas where there was none/null before.

It's a partial update because in reality there should be some reconciliation between all InChI/SMILE formulas and what we have in the database, but there's obviously going to be some problems and manual interpretation.

### CODE REVIEW

File: Libs/Python/BiochemPy/Compounds.py [hunk 1]
```diff
@@ -107,8 +107,7 @@ def loadStructures(self,sources_array=[],db_array=[]):
         
     @staticmethod
     def parseFormula(formula):
-        if (
-                        formula is None or formula == "" or "noFormula" in formula or "null" in formula):
+        if (formula is None or formula == "" or "noFormula" in formula or "null" in formula):
```
Comment (JamesJeffryes) [inline Libs/Python/BiochemPy/Compounds.py]: could achieve much the same with `if formula.strip() is in {None, "", "noFormula", "null"}:` assuming it is ok to match noFormula or null exactly.

File: Scripts/Biochemistry/Adjust_Reaction_Protons.py [hunk 2]
```diff
@@ -10,7 +10,11 @@
 Reactions_Dict = ReactionsHelper.loadReactions()
 
 Update_Reactions=0
+status_file = open("Status_Changes_After_Proton_Adjustment.txt",'w')
 for rxn in sorted(Reactions_Dict.keys()):
+#    if(rxn != "rxn00028"):
+#        continue
```
Comment (JamesJeffryes) [inline Scripts/Biochemistry/Adjust_Reaction_Protons.py]: can this be removed?

File: Scripts/Biochemistry/Check_Formulas.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    (inchi_formula,inchi_layers) = InChIs.parse(Structures_Dict[cpd]['InChI'])
+    adjusted_inchi_formula = InChIs.adjust_protons(inchi_formula,inchi_layers['p'])
+    if(adjusted_inchi_formula != Compounds_Dict[cpd]['formula']):
+        print cpd, inchi_formula, Compounds_Dict[cpd]['formula']
+        break
+
+
+
+#Update_Compounds=0
+#for cpd in sorted(Compounds_Dict.keys()):
```
Comment (JamesJeffryes) [inline Scripts/Biochemistry/Check_Formulas.py]: Can we remove this code?

File: Libs/Python/BiochemPy/Compounds.py (hunk 1 above)
Comment (samseaver) [reply Libs/Python/BiochemPy/Compounds.py]: if (formula.strip() is in {None, "", "noFormula", "null"}):
                            ^
SyntaxError: invalid syntax

**Pull request merged**